script.js

```diff
@@ -0,0 +1,80 @@
+// CHALLENGE 1
+// `// given an array of post objects create a function that returns a new array of posts, each object in the new array should have keys renamed as per the example. Original array of posts should not be mutated.`
+
+// `// @param {{id: number, createdBy: number, commentNo: number}[]}`
+
+// `// @returns {{postId: number, postCreatedBy: number }[]}`
+
+const postsArray = [
+  {
+    id: 123,
+    createdBy: "user16",
+    commentNo: 5,
+  },
+  {
+    id: 456,
+    createdBy: "user20",
+    commentNo: 5,
+  },
+  {
+    id: 789,
+    createdBy: "user32",
+    commentNo: 5,
+  },
+];
+
+const newPostArr = (arr) => {
+  [...arr].map((item) => {
+    return {
+      postsId: item.id,
+      postsCreated: item.createdBy,
+    };
+  });
```

Comment:
store what map returns in a variable and return that variable at the end of this function